comp/checks/winregistry/impl/integration_logs_registry_delegate.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2023-present Datadog, Inc.
+
+//go:build windows
+
+package winregistryimpl
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/comp/logs/agent"
+	"github.com/DataDog/datadog-agent/pkg/logs/message"
+	"time"
+)
+
+type integrationLogsRegistryDelegate struct {
+	baseRegistryDelegate
+	logsComponent agent.Component
+	valueMap      map[string]interface{}
+	origin        *message.Origin
+	muted         bool
+}
+
+const (
+	keyChanged = "key_changed"
+	keyCreated = "key_created"
+	keyDeleted = "key_deleted"
+)
+
+func (i *integrationLogsRegistryDelegate) sendLog(payload message.StructuredContent, status string) {
+	if i.logsComponent == nil || i.muted {
+		return
+	}
+	msg := message.NewStructuredMessage(payload, i.origin, status, time.Now().UnixNano())
+	i.logsComponent.GetPipelineProvider().NextPipelineChan() <- msg
+}
+
+func getPayload(m, eventType string, regKeyCfg registryKey) message.BasicStructuredContent {
+	payload := message.BasicStructuredContent{
+		Data: make(map[string]interface{}),
+	}
+	payload.Data["message"] = m
+	payload.Data["type"] = "windows_registry_event"
+	payload.Data["key_path"] = regKeyCfg.originalKeyPath
+	payload.Data["event_type"] = eventType
+
+	//payload.Data["process_id"] = coming in a future version
+	//payload.Data["process_name"] = coming in a future version
+	//payload.Data["thread_id"] = coming in a future version
+	return payload
+}
+
+func getKeyName(valueName string, regKeyCfg registryKey) string {
+	return fmt.Sprintf("%s\\%s", regKeyCfg.originalKeyPath, valueName)
+}
+
+func (i *integrationLogsRegistryDelegate) processValue(valueName string, val interface{}, regKeyCfg registryKey) {
+	keyName := getKeyName(valueName, regKeyCfg)
+	cachedVal, ok := i.valueMap[keyName]
+	if !ok {
+		p := getPayload(fmt.Sprintf("value %s = %v", valueName, val), keyCreated, regKeyCfg)
+		p.Data["old_value"] = nil
+		p.Data["new_value"] = val
+		i.sendLog(&p, "info")
+		i.valueMap[keyName] = val
+	} else {
+		if cachedVal != val {
+			p := getPayload(fmt.Sprintf("value %s change from %v to %v", valueName, cachedVal, val), keyChanged, regKeyCfg)
+			p.Data["old_value"] = cachedVal
+			p.Data["new_value"] = val
+			i.sendLog(&p, "info")
+			i.valueMap[keyName] = val
+		}
+	}
+}
+
+func (i *integrationLogsRegistryDelegate) onSendNumber(valueName string, val float64, regKeyCfg registryKey, _ registryValueCfg) {
+	i.processValue(valueName, val, regKeyCfg)
+}
+
+func (i *integrationLogsRegistryDelegate) onSendMappedNumber(valueName string, _ string, mappedVal float64, regKeyCfg registryKey, _ registryValueCfg) {
+	i.processValue(valueName, mappedVal, regKeyCfg)
+}
+
+func (i *integrationLogsRegistryDelegate) onNoMappingFound(valueName string, val string, regKeyCfg registryKey, _ registryValueCfg) {
+	i.processValue(valueName, val, regKeyCfg)
+}
+
+func (i *integrationLogsRegistryDelegate) onMissing(valueName string, regKeyCfg registryKey, _ registryValueCfg, _ error) {
+	if i.logsComponent == nil {
+		return
+	}
```

Comment:
Should we have this condition on all delegates? Should we even create a delegate to begin with